src/shared/proto/tbots_robot_msg.proto

```diff
@@ -0,0 +1,21 @@
+syntax = "proto2";
+
+import "shared/proto/tbots_robot_status_2014.proto";
+import "shared/proto/tbots_frankie_status.proto";
+
+message Tbots_RobotMsg {
+
+    // The robot id.
+    required int32 robot_id = 4;
+
+    // Messages related to the robot that are to be displayed in the visualizer
+    repeated string robot_messages = 5;
```

Comment:
strings are kind of annoying to use with `nanopb` (doable if needed, but annoying so it would be nice if we can avoid)

Is this so robots can send arbitrary debug messages over the network if needed? Or is this used for something else?

Reply:
I'm assuming this is legacy stuff that was moved from the old robot status structure. @jonathanlew I'm assuming you essentially copied most of the `RobotStatus` struct from `backend/robot_status.h` when creating these messages.

This old `RobotStatus` struct does contain strings, but they are added as part of the status handling in `backend/output/radio/mrf/annunciator.cpp`, in the `handle_robot_message` function on line `105`. This is where values are checked and if something isn't right, messages from a lookup table are added.

Eg.
```
new_msgs.insert(new_msgs.begin(), MRF::LOW_CAP_MESSAGE);
```

So to my knowledge the robots never sent string over radio, any strings that were added to the `RobotStatus` were from an error code or post-processing, so I don't think we need strings in the proto message.

Reply:
I see I'll change it to some sort of error code then

Reply:
We should maybe sync up about if we want to use predominantly error codes, or determine stuff with the values sent back (like we do currently in the annunciator).

Pros of error codes
* Should work better with a hybrid fleet. The software won't need to check raw values against different thresholds for different hardware
* Can get info / errors about things that aren't sent back explicitly (eg. SD Card errors)

Cons of error codes
* Can't get specific values for the errors. Eg. we can warn about low battery, but we don't know the exact value

IMO we should user error codes as much as possible for reporting faults and errors (primarily used to detect faulty robots for things like substitution, and to report in the Visualizer). All other values should just be for extra info, whether it be debugging or just readability (eg. battery status, cap voltage).

At least this way there's a clear distinction between where errors / faults come from, and where "data" comes from. The software should _not_ perform calculations on raw data to add additional errors.

Thoughts @garethellis0 @jonathanlew @akhilveeraghanta ?

Reply:
Sounds good.